### ISSUE TITLE
Memory very high and high CPU usage although TabNine local is disabled

### ISSUE THREAD

**Issue by manfredlotz** [codota/TabNine#215]
Running vim and tabnine on Fedora 31. 

Directly after start tabnine doesn't use much memory:

```
USER         PID %CPU %MEM    VSZ   RSS TTY
manfred    17596 2.8  0.4 284108 80516 ?        Sl
```

But after some hours it could happen that it really uses much memory and many cores.

```
USER         PID %CPU %MEM    VSZ   RSS TTY
manfred    17596  3.0 19.4 3424100 3173640 ?     Sl
```
Is there anything I could do to debug?

**Comment by manfredlotz** [comment]
Here some more observations. The very high CPU and memory usage was when I used `zxqfl/tabnine-vim`.

When using tabnine with deoplete, i.e. `tbodt/deoplete-tabnine` things seem to look better although memory also increases over time.

**Comment by Shougo** [comment]
You can disable Tabnine local to improve performance.

**Comment by manfredlotz** [comment]
I had disabled Tabnine local already. Performance is ok. Only memory increases over time. Nevertheless, with deoplete it is much better than with `tabnine-vim` which user `ycm`.

**Comment by drorweiss** [comment]
Hi @manfredlotz , we have a fix in beta (version 2.3.4) for excessive CPU and memory consumption by TabNine.
We'd appreciate it if you try it out and let us know if it solves the issue for you.
To try the fix, you have to enable the beta version. Just open TabNine's config by typing TabNine::config in your editor, and then check both **Update TabNine automatically** and **Receive beta releases of TabNine** Then, restart your editor, and press TabNine::confiig to verify you have TabNine 2.3.4. or newer.  Please let us know if this version solves the problem for you. We plan to make it available to all users in a couple of weeks.

**Comment by manfredlotz** [comment]
Hi @drorweiss, Thanks a lot.
I have done as you described, and will report in a couple of days.

**Comment by manfredlotz** [comment]
I have no real numbers but from looking at the memory consumption and such it is clear that TabNine's memory usage has improved and the peaks are lower than they were before. 

I noticed that it happens that memory usage goes also down over time. Not sure this happened before.

Memory usage is higher when doing Rust programming. Perhaps TabNine does more fancy things when it is about Rust. But here also the peaks are lower than they were before.


I am not sure about excessive CPU usage as I did not perceive it that much before.
